views/components/table.html

```diff
@@ -0,0 +1,22 @@
+{% macro table(data) %}
+<table class="base--table">
+  <caption class="base--caption">{{data.name}}</caption>
```

Comment:
As mentioned in my comments in https://github.com/punchcard-cms/punchcard/pull/555#issuecomment-261702309, this should be wrapped in an `if` to check to see if `data.name` exists, and only if it does should the caption be added